app/uk/gov/hmrc/agentclientrelationships/auth/AuthActions.scala

```diff
@@ -80,6 +81,39 @@ trait AuthActions extends AuthorisedFunctions {
   def hasRequiredStrideRole(enrolments: Enrolments, strideRoles: Seq[String]): Boolean =
     strideRoles.exists(s => enrolments.enrolments.exists(_.key == s))
 
+  def determineService(enrolment: Enrolment): Either[Unit, TypeOfEnrolment] =
+    enrolment.key match {
+      case EnrolmentMtdIt.enrolmentKey  => Right(EnrolmentMtdIt)
+      case EnrolmentMtdVat.enrolmentKey => Right(EnrolmentMtdVat)
+      case EnrolmentTrust.enrolmentKey  => Right(EnrolmentTrust)
+      case EnrolmentNino.enrolmentKey   => Right(EnrolmentNino)
+      case e =>
+        Logger(getClass).warn(s"Unsupported service: $e")
+        Left()
+    }
+
+  def withAuthorisedAsClientRequest[A](service: String)(body: Request[AnyContent] => TaxIdentifier => Future[Result])(
+    implicit ec: ExecutionContext): Action[AnyContent] = Action.async { implicit request =>
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    authorised(Enrolment(service) and AuthProviders(GovernmentGateway)).retrieve(allEnrolments) { enrolments =>
```

Comment:
do we need this here  `Enrolment(service)` ?  as we already doing `retrieve(allEnrolments)`

Reply:
ok